src/reconstruction/refinement/extendModelFieldsForType.m

```diff
@@ -55,44 +55,119 @@
-        %and that we don't have additional char arrays. logicals are 0 by
-        %default.
-        if iscell(model.(fields{field}))
-                model.(fields{field})(originalSize+1:targetSize,1) = {''};
-        elseif isnumeric(model.(fields{field})) 
-                model.(fields{field})(originalSize+1:targetSize,1) = NaN;
-        elseif islogical(model.(fields{field}))
-                model.(fields{field})(originalSize+1:targetSize,1) = false;
+        if isnumeric(model.(cfield))
+            fieldType = 'numeric';
+            defaultValue = 0;
         end
+        if islogical(model.(cfield))
+            fieldType = 'sparselogical';
+            defaultvalue = false;
+        end                
+    else
+        fieldType = cfieldDef{7};
+        defaultValue = cfieldDef{5};
+    end
+    cdim = dimensions(field);
+    fieldDims = size(model.(cfield));    
+    %Matlab will screw up multi-dimensional arrays when the are of size 1
+    %or zero in a dimension... So for now, we only handle two dimensional
+    %arrays.
+    %We need to handle completely empty arrays differently.
+    if(all(fieldDims == 0))
+        %This can only ever happen with numeric, logical or character fields!
+        %However, it is only relevant for numeric fields...
+        if isnumeric(model.(cfield))
+            if issparse(model.(cfield))
+                if cdim == 1
+                    model.(cfield) = sparse(targetSize,0);
+                else
+                    model.(cfield) = sparse(0,targetSize);
+                end
+            else
+                if cdim == 1
+                    model.(cfield) = zeros(targetSize,0);
+                else
+                    model.(cfield) = zeros(0,targetSize);
+                end
+            end
+            continue;
+        end
+    end
+    switch fieldType
+        case 'sparse'
+            model.(cfield) = extendIndicesInDimenion(model.(cfield),cdim,defaultValue, targetSize-originalSize);
+        case 'cell'
+            newValues = cell(0,1);
+            for i = originalSize+1:targetSize
+                eval(['currentvalue = ' defaultValue ';']);
+                newValues{end+1,1} = currentvalue;
+            end
+            model.(cfield) = extendIndicesInDimenion(model.(cfield),cdim,newValues, targetSize-originalSize);
+        case 'sparselogical'            
+            model.(cfield) = extendIndicesInDimenion(model.(cfield),cdim,logical(defaultValue), targetSize-originalSize);                        
+        case 'numeric'
+            model.(cfield) = extendIndicesInDimenion(model.(cfield),cdim,defaultValue, targetSize-originalSize);
+        case 'char'
+            model.(cfield) = extendIndicesInDimenion(model.(cfield),cdim,defaultValue, targetSize-originalSize);
     end
 end
-
 end
 
         
-        
\ No newline at end of file
+        
+
+function added = extendIndicesInDimenion(input,dimension,value, sizeIncrease)
```

Comment:
Would it be worth including this function as a separate function instead of an internal one?

Reply:
Hmm... could go into a utility. true